File: docs/manual/scala/guide/broker/code/docs/scaladsl/mb/AnotherService.scala
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.scaladsl.mb

import akka.NotUsed
import com.lightbend.lagom.scaladsl.api.Service
import com.lightbend.lagom.scaladsl.api.ServiceCall

trait AnotherService extends Service {

  final override def descriptor = {
    import Service._

    named("another-service")
      .withCalls(
        namedCall("/api/foo", foo)
      )
      .withAutoAcl(true)

  }

  def foo: ServiceCall[NotUsed, String]

}

File: docs/manual/java/guide/broker/code/docs/javadsl/mb/AnotherService.java
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.javadsl.mb;

import com.lightbend.lagom.javadsl.api.*;
import akka.NotUsed;

import static com.lightbend.lagom.javadsl.api.Service.*;

public interface AnotherService extends Service {
  ServiceCall<NotUsed, NotUsed> audit();

  @Override
  default Descriptor descriptor() {
    return named("anotherservice").withCalls(namedCall("/api/audit", this::audit));
  }
}
